Sources/Actions/Feed.php

```diff
@@ -774,7 +775,7 @@ public function getXmlNews(): array
 
 			// Limit the length of the message, if the option is set.
 			if (!empty(Config::$modSettings['xmlnews_maxlen']) && Utils::entityStrlen(str_replace('<br>', "\n", $row['body'])) > Config::$modSettings['xmlnews_maxlen']) {
-				$row['body'] = strtr(Utils::entitySubstr(str_replace('<br>', "\n", $row['body']), 0, Config::$modSettings['xmlnews_maxlen'] - 3), ["\n" => '<br>']) . '...';
+				$row['body'] = strtr(Utils::entitySubstr(str_replace('<br>', "\n", $row['body']), 0, (int) (Config::$modSettings['xmlnews_maxlen'] - 3)), ["\n" => '<br>']) . '...';
```

Comment:
Casting the result of the expression to int is unnecessary. The result of the arithmetic operation will already be an integer.

Reply:
Static analysis says it is a float.  I'm not certain where it gets a float from, I tried to work backwards to figure it out and gave up.

Reply:
Well, the static analysis is mistaken. I mean, the result theoretically _could_ be a float if Config::$modSettings['xmlnews_maxlen'] were a float. But it isn't.

Reply:
It doesn't know that though.  It knows `Config::`$modSettings` is an array with unknown members.  It doesn't know the member `xmlnews_maxlen` is an int.  Technically it isn't an int either, because we didn't cast it (I assume), so it's a string.  PHP is still loosely inferring it is an int by coercion.

Reply:
Then this would be a better approach:

```
				$row['body'] = strtr(Utils::entitySubstr(str_replace('<br>', "\n", $row['body']), 0, (int) Config::$modSettings['xmlnews_maxlen'] - 3), ["\n" => '<br>']) . '...';
```

Cast the original value to int rather than the result of expression.

Reply:
I pulled the changes out for now, so we can focus on the Sapi class.  I will send it up in another PR later.  Seems we need to have discussions on handling the strict typing for things like $modSettings which is causing an unknown int.  I wonder if I could just do it in the PHPDoc with `array|mixed[]|null` and appease the type hinting.